llvm/lib/Support/APFloat.cpp

```diff
@@ -99,6 +99,11 @@ enum class fltNanEncoding {
   NegativeZero,
 };
 
+class APFloatEBOChecker {
```

Comment:
You don't have to add a class for this assert -- it should work in the global scope.

Reply:
The reason is that `APFloat::U` is private, so we added a `friend` class to access it.

Reply:
OK, that's fair. Could you add a comment explaining this next to the friend declaration?

Reply:
Sure. Will do.